lib/noble.js

```diff
@@ -1,190 +1,129 @@
-    try {
-      const res = bitcoinerlab.privateAdd(privateKey, tweak)
-      if (!res) return 1
-      privateKey.set(res)
-    } catch (e) {
-      console.error('privateKeyTweakAdd failed', e)
-      return 1
-    }
-    return 0
-  },
-  privateKeyTweakMul: function (seckey, tweak) {
-    try {
-      const keyNum = noble.utils._normalizePrivateKey(seckey)
-      const tweakNum = noble.utils._normalizePrivateKey(tweak)
-      const res = noble.utils.mod(keyNum * tweakNum, noble.CURVE.n)
-      seckey.set(noble.utils._bigintTo32Bytes(res))
-    } catch (e) {
-      console.error('privateKeyTweakMul failed', e)
-      return 1
-    }
-    return 0
-  },
+  ...elliptic,
   publicKeyVerify: function (publicKey) {
-    return bitcoinerlab.isPoint(publicKey) ? 0 : 1
+    return _isPoint(publicKey) ? 0 : 1
   },
   publicKeyCreate: function (output, seckey) {
     try {
-      const res = bitcoinerlab.pointFromScalar(
-        seckey,
-        isCompressedPublicKey(output)
-      )
-      if (!res) return 2
-      output.set(res)
-      return 0
+      output.set(noble.getPublicKey(seckey, isCompressedPublicKey(output)))
     } catch (e) {
       console.error('publicKeyCreate failed', e)
       return 1
     }
+    return 0
   },
   publicKeyConvert: function (output, pubkey) {
     try {
       output.set(
-        bitcoinerlab.pointCompress(pubkey, isCompressedPublicKey(output))
+        noble.Point.fromHex(pubkey).toRawBytes(isCompressedPublicKey(output))
       )
-      return 0
     } catch (e) {
       console.error('publicKeyConvert failed', e)
-      if (e.message === errors.EXPECTED_POINT) return 1
-      return 2
-    }
-  },
-  publicKeyNegate: function (output, pubkey) {
-    try {
-      output.set(
-        noble.Point.fromHex(pubkey)
-          .negate()
-          .toRawBytes(isCompressedPublicKey(output))
-      )
-    } catch (e) {
-      console.error('publicKeyNegate failed', e)
-      return 1
-    }
-    return 0
-  },
-  publicKeyCombine: function (output, pubkeys) {
-    try {
-      let point = noble.Point.fromHex(pubkeys[0])
-      for (let i = 1; i < pubkeys.length; ++i) {
-        point = point.add(noble.Point.fromHex(pubkeys[i]))
-      }
-      try {
-        point.assertValidity()
-      } catch (err) {
-        console.error('publicKeyCombine failed', err)
-        return 2
-      }
-      output.set(point.toRawBytes(isCompressedPublicKey(output)))
-    } catch (e) {
-      console.error('publicKeyCombine failed', e)
       return 1
     }
     return 0
   },
   publicKeyTweakAdd: function (output, pubkey, tweak) {
     try {
-      const res = bitcoinerlab.pointAddScalar(
-        pubkey,
-        tweak,
-        isCompressedPublicKey(output)
-      )
-      if (!res) return 2
-      output.set(res)
-      return 0
+      const P = noble.Point.fromHex(pubkey)
+      const t = normalizeScalar(tweak)
+      const Q = noble.Point.BASE.multiplyAndAddUnsafe(P, t, BigInt(1))
```

Comment:
```suggestion
      const T = noble.fromPrivateKey(tweak)
      const Q = P.add(T)
```

Maybe less scary 😁

Reply:
The original code is safe. The 'unsafe' in the name of that method just means 'non-constant time', which is OK in this use case because the tweak is considered public data.

https://github.com/ExodusMovement/secp256k1-node/pull/7/files/bfca9111fa2eccb1523fcad29d74c97817622c22#r1476856621